src/data_processing/wiki_to_netflix.py

```diff
@@ -0,0 +1,136 @@
+    wiki_movie_ids = []
+    wiki_genres = []
+    wiki_directors = []
+    
+    # csv_reader = list(csv.reader(open(data_csv)))
+    
+    for row in data_csv:
+        if row[1] != "NULL":
+            SPARQL = '''
+            SELECT * WHERE {
+                SERVICE wikibase:mwapi {
+                    bd:serviceParam wikibase:api "EntitySearch" ;
+                                    wikibase:endpoint "www.wikidata.org" ;
+                                    mwapi:search "%(Title)s" ;
+                                    mwapi:language "en" .
+                    ?item wikibase:apiOutputItem mwapi:item .
+                }
+
+                ?item wdt:P31/wdt:P279* wd:Q11424 .
+                
+                {
+                    # Get US release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    ?releaseDateStatement pq:P291 wd:Q30 .  
+                }
+                UNION
+                {
+                    # Get unspecified release date
+                    ?item p:P577 ?releaseDateStatement .
+                    ?releaseDateStatement ps:P577 ?releaseDate .
+                    FILTER NOT EXISTS { ?releaseDateStatement pq:P291 ?country }
+                }
+            
+                FILTER (YEAR(?releaseDate) = %(Year)d) .
+
+                ?item rdfs:label ?itemLabel .
+                FILTER (lang(?itemLabel) = "en") .
+
+                OPTIONAL {
+                    ?item wdt:P136 ?genre .
+                    ?genre rdfs:label ?genreLabel .
+                    FILTER (lang(?genreLabel) = "en") .
+                }
+
+                OPTIONAL {?item wdt:P57 ?director.
+                                ?director rdfs:label ?directorLabel.
+                                FILTER (lang(?directorLabel) = "en")}
+
+                SERVICE wikibase:label { bd:serviceParam wikibase:language "en" . }
+                }
+        
+            ''' % {'Title': row[2], 'Year': int(row[1])}
+
+            response = requests.post('https://query.wikidata.org/sparql',
+                      headers={'User-Agent': user_agent},
+                      data={
+                        'query': SPARQL,
+                        'format': 'json',
+                      }
+        )
+            response.raise_for_status() 
+            
+            data = response.json()
+            
+            wiki_movie_ids.append(wiki_feature_info(data, 'item'))
+            wiki_genres.append(wiki_feature_info(data, 'genreLabel'))
+            wiki_directors.append(wiki_feature_info(data, 'directorLabel'))
+        
+    return(wiki_movie_ids, wiki_genres, wiki_directors)
+
+# Calling all functions
+processed_data = []
+
+netflix_file = read_netflix_txt(base_dir + 'netflix_movies.txt')
+
+netflix_csv = base_dir + 'netflix_movies.csv'
+
+wiki_movie_ids_list, wiki_directors_list, wiki_genres_list = wiki_query(netflix_file, user_agent)
```

Comment:
It looks like you may have reversed some fields here? On line 114 it says you're returning ids, genres and directors. But here you're assigning ids, directors then genres.

Reply:
Yeah I think it got mixed up. Should be fixed